src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpLanguageCharacteristics.cs

```diff
@@ -64,8 +62,104 @@ internal class CSharpLanguageCharacteristics : LanguageCharacteristics<CSharpTok
             { SyntaxKind.Transition, "@" },
         };
 
+        private static Dictionary<CSharpKeyword, string> _keywordNames = new Dictionary<CSharpKeyword, string>()
```

Comment:
```suggestion
        private static IReadOnlyDictionary<CSharpKeyword, string> _keywordNames = new Dictionary<CSharpKeyword, string>()
```

Reply:
Add a comment as to why we have this huge list. It'll be confusing for future onlookers without one.

Reply:
Done